src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,11 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user)
+                .map(user1 -> user1.getAge())
+                .filter(age -> age >= 30)
+                .filter(age -> age <= 45)
+                .isPresent();
```

Comment:
```suggestion
        return Optional.ofNullable(user)
                .filter(User::isInRange)
                .isPresent();
```

의미있는 이름을 가진 메소드명과 함수형 프로그래밍을 함께 사용한다면 시너지가 있을것 같네요 🤔